### DISCUSSION TITLE
ruTorrent v5.1 roadmap

### DISCUSSION THREAD

**Discussion by stickz** [Novik/ruTorrent#2740] [General]
Do you have any thoughts about this plan @jevenski or @koblack ?

**Stage 1 - Bug Fixes**
- [x] Fix the table sorting regression introduced in version 5. #2735
- [x] Release ruTorrent v5.1 - Beta 2

**Stage 2 - Other Changes**
- [x] Limited context menu improvements for v5.1. #2734
- [x] Persistent add and create torrent options for v5.1. #2728
- [x] Update fonts for material design theme for v5.1 #2736 
- [x]  Address non existent function for v5.1. #2751
- [x] Release ruTorrent v5.1 - Beta 3 https://github.com/Novik/ruTorrent/releases/tag/v5.1-beta3

**Stage 3 - Bug Fixes**
- [x] Address any other issues brought to our attention.
- [x] Release ruTorrent v5.1 - Stable

**Comment by jevenski** [comment]
For a moment, I was thinking were we going too fast on rolling out new releases? Looking back at the release history, it can be seen that it took almost a year to go from v4.0's first beta to its stable release. Although we don't need as long as one year again for v5, maybe we still need to take some time to fully test the APP, catch bugs, and nail them one by one.

As for the navbar part, I was planning to address it tegother with increasing the font size for mobile, as this change will make the buttons a bit larger and also affect the layout of the navbar. But if the limited context menu improvement is more urgent, I'm okay with doing that first.

**Reply by stickz** [reply]
These 3 things are urgent to get us to ruTorrent `v5.1`. The planned outcome is to mass deploy this release to ten thousand people. One thousand people from whatbox are very happy with the results. We just need a little bit more polish then everyone will adopt it!

We can increase the font size in 1-2 weeks, when we start a new development cycle for ruTorrent `v5.2`! The web client is more stable than it was years ago, even with the accelerated development cycles! We're nailing our agile sprints with the release control steps being taken!

**Reply by jevenski** [reply]
Yeah, it's surely a significant milestone. I was saying so because so far I've only been playing with the JS part of ruTorrent, and remembering torrent options from last time requires some work on the PHP side. If I were to modify that part, I need some time to get familiar with the PHP codebase first before doing anything. That might slow down the process. Of course it would be appreciated if someone else can take over this part.

**Reply by stickz** [reply]
> Yeah, it's surely a significant milestone. I was saying so because so far I've only been playing with the JS part of ruTorrent, and remembering torrent options from last time requires some work on the PHP side. If I were to modify that part, I need some time to get familiar with the PHP codebase first before doing anything. That might slow down the process. Of course it would be appreciated if someone else can take over this part.

Can you put the implementation in JS for now? Anything in webui settings will be saved. It's also threaded. I wrote this myself!

**Comment by jevenski** [comment]
Some thoughts on features of v5.2 (mostly front-end related):

1. Increase font size to 16px on mobile. This will greatly affect the layout on the entire APP, so I guess many issues will arise and fundamental adjustments will be required.
2. Define theme colors using CSS variables, in order to reduce duplication in themes. This way maintaining themes can become a lot easier and less error prone. Maybe we can drop some built-in themes, for example Excel (it looks like Excel 2007 and it's 2024 now), and Dark (we already have a better version of it).
3. Remove the `dragMask` object so that we can drag and move dialog windows instantly, instead of moving a "shadow" window around on the go and relocate the actual window to that place. This has already been implemented on the resize-bar. We can remove the drag mask on `DnD` class completely in v5.2.

**Reply by stickz** [reply]
1. Yes we can adjust font sizes in `v5.2` and address any issues which come up. We shouldn't be dead set on 16px though. We need to test to find the best value to replace 11px.

2. We can drop the `Dark` theme in `v5.2` as the `DarkBetter` replacement is now in better shape. I would like to keep the classic Excel theme though because it's different than the new version.

3. Yes can we remove the `dragMask` object in `v5.2`. I don't think this feature serves any useful purpose
